### ISSUE TITLE
selenium / chromedriver not working on OSX

### ISSUE THREAD

**Issue by kleinjm** [mrooney/mintapi#95]
After a successful install of mintapi, chromedriver, and selenium, I see this output.
```
➜  dashboard git:(master) ✗ mintapi
Mint e-mail: xxx@xxx.com <- I put my real email here
Mint password:
Traceback (most recent call last):
  File "/usr/local/bin/mintapi", line 9, in <module>
    load_entry_point('mintapi==1.23', 'console_scripts', 'mintapi')()
  File "/Library/Python/2.7/site-packages/mintapi/api.py", line 703, in main
    mint = Mint.create(email, password, ius_session=options.session, thx_guid=options.thx_guid)
  File "/Library/Python/2.7/site-packages/mintapi/api.py", line 74, in create
    mint.login_and_get_token(email, password, ius_session, thx_guid)
  File "/Library/Python/2.7/site-packages/mintapi/api.py", line 135, in login_and_get_token
    session_cookies = self.get_session_cookies(**data)
  File "/Library/Python/2.7/site-packages/mintapi/api.py", line 183, in get_session_cookies
    (e.args[0] if len(e.args) > 0 else 'No error message found.'))
RuntimeError: ius_session not specified, and was unable to load the chromedriver selenium plugin. Please ensure that the `selenium` and `chromedriver` packages are installed.

The original error message was: No error message found.
```
`brew services list` ->
```
chromedriver      started ... /Users/.../Library/LaunchAgents/homebrew.mxcl.chromedriver.plist
```

**Comment by mrooney** [comment]
Thanks for the report! Can you try running this from a terminal and let me know the output?

`python -c "from selenium import webdriver; driver = webdriver.Chrome()"`

That's what is happening under the hood where that exception is being caught, so something isn't working with one of those lines.

**Comment by kleinjm** [comment]
Thanks for the reply, this is what I'm seeing.
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Library/Python/2.7/site-packages/selenium/webdriver/chrome/webdriver.py", line 62, in __init__
    self.service.start()
  File "/Library/Python/2.7/site-packages/selenium/webdriver/common/service.py", line 96, in start
    self.assert_process_still_running()
  File "/Library/Python/2.7/site-packages/selenium/webdriver/common/service.py", line 109, in assert_process_still_running
    % (self.path, return_code)
selenium.common.exceptions.WebDriverException: Message: Service chromedriver unexpectedly exited. Status code was: 127
```

**Comment by mrooney** [comment]
Interesting, you are on OSX it seems, as I am. Have you tried restarting the computer? Is this a headless machine that doesn't have a logged in graphical user, by chance?

**Comment by alizrrao** [comment]
Also, for anyone running this on Ubuntu, running

`python -c "from selenium import webdriver; driver = webdriver.Chrome()"`

may give you the error:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/chrome/webdriver.py", line 62, in __init__
    self.service.start()
  File "/usr/local/lib/python2.7/dist-packages/selenium/webdriver/common/service.py", line 81, in start
    os.path.basename(self.path), self.start_error_message)
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://sites.google.com/a/chromium.org/chromedriver/home

Exception AttributeError: "'Service' object has no attribute 'process'" in <bound method Service.__del__ of <selenium.webdriver.chrome.service.Service object at 0x7f6080bc1210>> ignored
```

In this case, you will need to add the location of chromedriver to your PATH:

`export PATH=$PATH:/usr/lib/chromium-browser/`

If that doesn't solve the error (it didn't for me), you will need to add the path to Chromedriver to api.py, as well, on line 177. E.g.:

```
from selenium import webdriver
driver = webdriver.Chrome('/usr/lib/chromium-browser/chromedriver')
```

**Comment by kleinjm** [comment]
@mrooney sorry for the delay, haven't had much time to play with this. I've restarted and I'm getting a different error now that seems to be a type error. To answer your other question, no this is not a headless machine that I'm on. It's out of the box Sierra 10.12.5.
```
➜  dashboard git:(master) ✗ mintapi
Mint e-mail: xxx@xxx.com
Mint password:
Traceback (most recent call last):
  File "/usr/local/bin/mintapi", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/mintapi/api.py", line 701, in main
    mint = Mint.create(email, password, ius_session=options.session, thx_guid=options.thx_guid)
  File "/usr/local/lib/python2.7/site-packages/mintapi/api.py", line 74, in create
    mint.login_and_get_token(email, password, ius_session, thx_guid)
  File "/usr/local/lib/python2.7/site-packages/mintapi/api.py", line 127, in login_and_get_token
    self.request_and_check(login_url)
  File "/usr/local/lib/python2.7/site-packages/mintapi/api.py", line 107, in request_and_check
    result = getattr(self, method)(url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 536, in get
    return self.request('GET', url, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 523, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/usr/local/lib/python2.7/site-packages/urllib3/connectionpool.py", line 587, in urlopen
    timeout_obj = self._get_timeout(timeout)
  File "/usr/local/lib/python2.7/site-packages/urllib3/connectionpool.py", line 302, in _get_timeout
    return Timeout.from_float(timeout)
  File "/usr/local/lib/python2.7/site-packages/urllib3/util/timeout.py", line 154, in from_float
    return Timeout(read=timeout, connect=timeout)
  File "/usr/local/lib/python2.7/site-packages/urllib3/util/timeout.py", line 94, in __init__
    self._connect = self._validate_timeout(connect, 'connect')
  File "/usr/local/lib/python2.7/site-packages/urllib3/util/timeout.py", line 127, in _validate_timeout
    "int, float or None." % (name, value))
ValueError: Timeout value connect was Timeout(connect=None, read=None, total=None), but it must be an int, float or None.
```

**Comment by colej4586** [comment]
@alizrrao , I'm also running into the same issue on Ubuntu server 1604. Your fixes unfortunately did not work for me. 

`Traceback (most recent call last):
  File "/usr/local/bin/mintapi", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/mintapi/api.py", line 701, in main
    mint = Mint.create(email, password, ius_session=options.session, thx_guid=options.thx_guid)
  File "/usr/local/lib/python2.7/dist-packages/mintapi/api.py", line 74, in create
    mint.login_and_get_token(email, password, ius_session, thx_guid)
  File "/usr/local/lib/python2.7/dist-packages/mintapi/api.py", line 135, in login_and_get_token
    session_cookies = self.get_session_cookies(**data)
  File "/usr/local/lib/python2.7/dist-packages/mintapi/api.py", line 183, in get_session_cookies
    (e.args[0] if len(e.args) > 0 else 'No error message found.'))
RuntimeError: ius_session not specified, and was unable to load the chromedriver selenium plugin. Please ensure that the `selenium` and `chromedriver` packages are installed.`

**Comment by vesper8** [comment]
I too was having this error on OSX Sierra and I just realized I simply hadn't installed Chromium itself since the instructions to install chromdriver and selenium don't mention installing Chromium like so: ```brew cask install chromium```

unfortunately running the interactive mintapi I am getting this error after it is done with auth and flashes the account screen 
```
Traceback (most recent call last):
  File "/usr/local/bin/mintapi", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/site-packages/mintapi/api.py", line 708, in main
    mint = Mint.create(email, password, ius_session=options.session, thx_guid=options.thx_guid)
  File "/usr/local/lib/python3.6/site-packages/mintapi/api.py", line 74, in create
    mint.login_and_get_token(email, password, ius_session, thx_guid)
  File "/usr/local/lib/python3.6/site-packages/mintapi/api.py", line 135, in login_and_get_token
    session_cookies = self.get_session_cookies(**data)
  File "/usr/local/lib/python3.6/site-packages/mintapi/api.py", line 199, in get_session_cookies
    'ius_session': driver.get_cookie('ius_session')['value'],
TypeError: 'NoneType' object is not subscriptable
```

**Comment by russlooker** [comment]
I just tried installing chromium on a new OSX sierra install and got the following error:
```Traceback (most recent call last):
  File "mint_etl.py", line 268, in <module>
    myInstance = executionManager()
  File "mint_etl.py", line 172, in __init__
    globalExecutionData()       #Instantiate Global runtime data
  File "mint_etl.py", line 80, in __init__
    self.mint = mintapi.Mint('russelljgarner@gmail.com', '&tlnplx07A')
  File "/Users/Russell/anaconda/lib/python2.7/site-packages/mintapi/api.py", line 69, in __init__
    self.login_and_get_token(email, password, ius_session, thx_guid)
  File "/Users/Russell/anaconda/lib/python2.7/site-packages/mintapi/api.py", line 135, in login_and_get_token
    session_cookies = self.get_session_cookies(**data)
  File "/Users/Russell/anaconda/lib/python2.7/site-packages/mintapi/api.py", line 199, in get_session_cookies
    'ius_session': driver.get_cookie('ius_session')['value'],
TypeError: 'NoneType' object has no attribute '__getitem__'```

**Comment by mrooney** [comment]
The window pops in the background usually, but if you click it and watch, there's often a specific thing you need to resolve in this case, like a captcha or similar. What does the Chrom(e|ium) window say in this case? PS, you shouldn't need Chromium if you have Google Chrome installed, FWIW.

**Comment by mrooney** [comment]
Give it a shot with the latest version from Github, with #150 where Chrome driver suppport is improved, including headless and MFA support.

**Comment by jaron-l** [comment]
I know this is an old issue, but I ran into the problem again. I'm running mintapi in a container based on alpine linux. I was confused at why it couldn't find my cromedriver (as I've run other selenium tests that have worked fine) until I saw where you create the Chrome webdriver instance in api.py in [get_web_driver](https://github.com/mrooney/mintapi/blob/4bf3f6fe6b90005b4e2b0b63c690786b869bfc4d/mintapi/api.py#L188-L216). You only look for chromedriver in the current directory and then attempt to download the current version if it's not found. It might be better just to initially test if it's in the path as I think your chromedriver request is getting the wrong chromedriver for me. I suspect this reasoning is why it might be failing for others (wrong chromedriver). If you want I could submit a pull request later with code for checking the path and then maybe an attempt at getting the chromedriver. It looks like the chromedriver version you're pulling is hard-coded in and may not work for all systems. Not 100% what a better way might be yet...
